MailCore/Models/Folder.swift

```diff
@@ -178,8 +178,7 @@ public class Folder: Object, Codable, Comparable, Identifiable {
     }
 
     public func computeUnreadCount(using realm: Realm) {
-        let unreadMessages = realm.objects(Message.self).where { $0.folderId == self.id && $0.seen == false }
-        unreadCount = unreadMessages.count
+        unreadCount = threads.where { $0.hasUnseenMessages == true }.count
```

Comment:
This seems wrong, if I have a thread with 2 messages marked unread it will still display 1 unread in the header ?

Reply:
Yes it looks like this counts the number of threads with unread messages in them, is this what is expected?